File: 0810-61-2948-make-lexicographically-smallest-array-by-swapping-elements/0810-61-2948-make-lexicographically-smallest-array-by-swapping-elements.cpp
class Solution {
public:
    vector<int> lexicographicallySmallestArray(vector<int>& nums, int limit) {
        int n = nums.size();
        vector<pair<int,int>>v;
        for(int i = 0 ; i < n;i++)v.push_back({nums[i],i});
        sort(v.begin(),v.end());

        for(int i = 0 ; i < n;){
            vector<int>idx;
            idx.push_back(v[i].second);
            int j = i + 1;
            while( j < n && v[j].first - v[j-1].first <= limit)idx.push_back(v[j].second),j++;
            sort(idx.begin(),idx.end());
            for(int k = 0 ; k < idx.size();k++)nums[idx[k]] = v[i+k].first;
            i = j;
        }
        return nums;


    }
};

File: 0810-61-2948-make-lexicographically-smallest-array-by-swapping-elements/0810-61-2948-make-lexicographically-smallest-array-by-swapping-elements.js
var lexicographicallySmallestArray = function(nums, limit) {
    const n = nums.length;
    const v = nums.map((val, i) => [val, i]).sort((a, b) => a[0] - b[0]);

    let i = 0;
    while (i < n) {
        const idx = [v[i][1]];
        let j = i + 1;
        while (j < n && v[j][0] - v[j - 1][0] <= limit) {
            idx.push(v[j][1]);
            j++;
        }
        idx.sort((a, b) => a - b);
        for (let k = 0; k < idx.length; k++) {
            nums[idx[k]] = v[i + k][0];
        }
        i = j;
    }

    return nums;
};
